tests/contrib/wsgi/test_wsgi.py

```diff
@@ -0,0 +1,71 @@
+import pytest
+from webtest import TestApp
+
+from ddtrace.vendor import six
+
+from ddtrace.compat import PY2, PY3
+from ddtrace.contrib.wsgi import wsgi
+
+from tests import snapshot
+
+
+def chunked_response():
+    for i in range(1000):
+        yield "%d" % i
+
+
+def application(environ, start_response):
+    if environ["PATH_INFO"] == "/error":
+        raise Exception("Oops!")
+    elif environ["PATH_INFO"] == "/chunked":
+        return chunked_response()
+    else:
+        body = six.b("<html><body><h1>Hello World</h1></body></html>")
+        headers = [("Content-Type", "text/html; charset=utf8"), ("Content-Length", str(len(body)))]
+        start_response("200 OK", headers)
+        return [body]
+
+
+def test_middleware(tracer):
+    app = TestApp(wsgi.DDWSGIMiddleware(application, tracer=tracer))
+    resp = app.get("/")
+    assert resp.status == "200 OK"
+    assert resp.status_int == 200
+    spans = tracer.writer.pop()
+    assert len(spans) == 4
+
+    with pytest.raises(Exception):
+        app.get("/error")
+
+    spans = tracer.writer.pop()
+    assert len(spans) == 2
+    assert spans[0].error == 1
+
+
+def test_distributed_tracing(tracer):
+    app = TestApp(wsgi.DDWSGIMiddleware(application, tracer=tracer))
+    resp = app.get("/", headers={"X-Datadog-Parent-Id": "1234", "X-Datadog-Trace-Id": "4321"})
+    assert resp.status == "200 OK"
+    assert resp.status_int == 200
+
+    spans = tracer.writer.pop()
+    assert len(spans) == 4
+    root = spans[0]
+    assert root.name == "wsgi.request"
+    assert root.trace_id == 4321
+    assert root.parent_id == 1234
+
+
+@snapshot()
+def test_200():
+    app = TestApp(wsgi.DDWSGIMiddleware(application))
+    resp = app.get("/")
+    assert resp.status == "200 OK"
+    assert resp.status_int == 200
+
+
```

Comment:
Some other test cases to add

- [ ] configuring service name 
- [ ] query string
- [ ] request headers
- [ ] response headers